### PR TITLE
NAS-3868 extend user and organisation service 

### PR DESCRIPTION
**Pull request by hkad98**
* added method update_oidc_parameters in organization service
* changed create to create_or_update for users and user groups
* changed storing users and user groups to layout
* added test support

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

File: gooddata-sdk/tests/catalog/test_catalog_organization.py [hunk 1]
```diff
@@ -19,5 +19,45 @@ def test_get_organization(test_config):
...
+    )
+
+    updated_organization = sdk.catalog_organization.get_organization()
+    assert organization.id == default_organization_id
+    assert organization.attributes.name == default_organization_name
+    assert organization.attributes.hostname == default_organization_hostname
+    assert updated_organization.attributes.oauth_issuer_location == oauth_issuer_location
+    assert updated_organization.attributes.oauth_client_id == oauth_client_id
+
+    sdk.catalog_organization.update_oidc_parameters(
```
Comment (jaceksan) [inline gooddata-sdk/tests/catalog/test_catalog_organization.py:55]: I would call the previous code with "try except" to ensure that this revert happens every time.

File: gooddata-sdk/gooddata_sdk/catalog/organization/service.py [hunk 2]
```diff
@@ -1,10 +1,30 @@
...
 from gooddata_sdk.catalog.catalog_service_base import CatalogServiceBase
+from gooddata_sdk.catalog.organization.entity_model.organization import CatalogOrganizationDocument
 from gooddata_sdk.client import GoodDataApiClient
 
 
 class CatalogOrganizationService(CatalogServiceBase):
     def __init__(self, api_client: GoodDataApiClient) -> None:
         super(CatalogOrganizationService, self).__init__(api_client)
+
+    def update_oidc_parameters(
```
Comment (jaceksan) [inline gooddata-sdk/gooddata_sdk/catalog/organization/service.py:15]: Unless it is too expensive, please, implement additional methods for updating "name" and "hostname", both are valid business cases.
Keep the new methods separate, they are different use cases.

**Review by jaceksan — CHANGES_REQUESTED** [review]

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

File: gooddata-sdk/tests/catalog/test_catalog_organization.py (hunk 1 above)
Comment (hkad98) [reply gooddata-sdk/tests/catalog/test_catalog_organization.py]: Done.

**Review by jaceksan — APPROVED** [review]

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-python-sdk-gate-PR108

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-python-sdk-sonarqube-scan-zuul-gate/100/ : SUCCESS in 1m 29s
- https://checklist.intgdc.com/job/compliance/job/gooddata-python-sdk-fossa-scanning-zuul-el7/100/ : SUCCESS in 3m 09s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-python-sdk-merge-compliance-check/100/ : SUCCESS in 23s

**Pull request merged**